src/common/include/gudhi/reader_utils.h

```diff
@@ -305,8 +305,12 @@ void read_persistence_intervals_and_dimension(std::string const& filename, Outpu
     std::string line;
     getline(in, line);
     if (line.length() != 0 && line[0] != '#') {
-      double numbers[4];
+      double numbers[4] = {0.};
```

Comment:
Is it because of one of gcc's idiotic warnings?

Reply:
I add a weird bug and could not find an explanation. I tried to set the double array before filling it, but it did not help.
I did not remove it, but I can do it.

Reply:
If it does not help for anything, then yes, I'd rather not add the initialization.
Is the weird bug still there?

Reply:
Yes it is still there, even with the master branch. It does not seem to come from the development done in this branch

Reply:
I removed the initialization on 57c1f0f4b36b934c8fc54c8691f77b9b6d0eff9d

Reply:
Ça ne colle pas bien avec le DEBUG_TRACES en dessous. Il faudrait soit que le code dans DEBUG_TRACES ne lise que les n premiers éléments, soit qu'on vire ce code, soit qu'on initialise (avec 4 qNaN par exemple ?).

Reply:
> Yes it is still there, even with the master branch. It does not seem to come from the development done in this branch

ok, then file an issue and let's ignore it for this PR.

Reply:
Ok.
I reworked traces on 1e3b4f6c56409dde933341701575dc105ede12db
The issue is tracked by #111